src/arch/xtensa/hal/CMakeLists.txt

```diff
@@ -133,27 +133,32 @@ target_compile_definitions(hal PRIVATE
 )
 
 add_local_sources(hal
-	attribute.c
-	cache.c
 	cache_asm.S
 	clock.S
-	coherence.c
-	debug.c
-	debug_hndlr.S
-	disass.c
 	int_asm.S
 	interrupts.c
 	memcopy.S
-	mem_ecc_parity.S
-	misc.c
-	miscellaneous.S
-	mmu.c
-	mp_asm.S
-	mpu_asm.S
-	mpu.c
-	set_region_translate.c
-	state_asm.S
-	state.c
-	syscache_asm.S
 	windowspill_asm.S
 )
+
+if (NOT ${ARCH} STREQUAL "xtensa")
```

Comment:
ok, if they are not used by any other rules in the HAL makefiles (e.g. for different configurations - this may have been true in the past but not now) then I would drop this part too as it looks confusing. Just keep the files in the directory and remove from our Makefile.

Reply:
@lgirdwood wouldn't it be awkward to have "dangling" files around not included in a build? Maybe add some "compile test" Kconfig?

Reply:
Usually yes, but these files all come from the xtensa HAL package, so will be expected by users.